ddtrace/tracer.py

```diff
@@ -738,8 +738,7 @@ def execute():
         """
 
         def wrap_decorator(f):
-            # FIXME[matt] include the class name for methods.
-            span_name = name if name else "%s.%s" % (f.__module__, f.__name__)
+            span_name = name or "%s.%s" % (f.__module__, f.__qualname__)
```

Comment:
This is a breaking change for resource names so I think we'll want to have this behind a configuration option so that users can fallback to the previous functionality.


```suggestion
            if config.fully_qualified_func_names:
                span_name = name or "%s.%s" % (f.__module__, f.__qualname__)
            else:
                span_name = name if name else "%s.%s" % (f.__module__, f.__name__)	
```


The config option would have to be defined here: https://github.com/DataDog/dd-trace-py/blob/387abf138cad7b432f510ca0598a5ef9dcb80166/ddtrace/settings/config.py#L61